packages/application-generic/src/usecases/get-subscriber-template-preference/get-subscriber-template-preference.usecase.spec.ts

```diff
@@ -0,0 +1,180 @@
+import {
+  determineOverrides,
+  filteredPreference,
+} from './get-subscriber-template-preference.usecase';
+import { ChannelTypeEnum } from '@novu/shared';
+
+describe('determineOverrides', function () {
+  beforeEach(function () {});
+
+  it('should be overridden by the subscribers preference', async function () {
+    const templateChannelPreference = {
+      email: false,
+      sms: true,
+      in_app: true,
+      chat: true,
+      push: true,
+    };
+    const subscriberChannelPreference = {
+      email: true,
+      sms: true,
+      push: false,
+    };
+
+    const { preferences, overrides } = determineOverrides(
+      {
+        template: templateChannelPreference,
+        subscriber: subscriberChannelPreference,
+      },
+      {
+        email: true,
+        sms: true,
+        in_app: true,
+        chat: true,
+        push: true,
+      }
+    );
+
+    const expectedPreferenceResult = {
+      email: true,
+      sms: true,
+      in_app: true,
+      chat: true,
+      push: false,
+    };
+
+    expect(preferences).toEqual(expectedPreferenceResult);
+    expect(
+      overrides.find((override) => override.channel === 'email').source
+    ).toEqual('subscriber');
+    expect(
+      overrides.find((override) => override.channel === 'sms').source
+    ).toEqual('subscriber');
+    expect(
+      overrides.find((override) => override.channel === 'in_app').source
+    ).toEqual('template');
+    expect(
+      overrides.find((override) => override.channel === 'chat').source
+    ).toEqual('template');
+    expect(
+      overrides.find((override) => override.channel === 'push').source
+    ).toEqual('subscriber');
+  });
+
+  it('should get preference from template when subscriber preference are empty', async function () {
```

Comment:
we're missing the test when the template preferences are empty, shouldn't we have one?